packages/flutter/lib/src/material/ink_sparkle.dart

```diff
@@ -0,0 +1,3658 @@
+    await manager.compile();
+    return manager;
+  }
+
+  /// Compiles the spir-v bytecode into a shader program.
+  Future<void> compile() async {
+    _program = await ui.FragmentProgram.compile(spirv: spirvByteBuffer);
+  }
+
+  /// Creates a shader with the original program and optional uniforms.
+  ///
+  /// A new shader must be made whenever the uniforms are updated.
+  Shader shader({
+    Vector4? uColor,
+    double? uAlpha,
+    Vector4? uSparkleColor,
+    double? uSparkleAlpha,
+    double? uBlur,
+    Vector2? uCenter,
+    double? uRadiusScale,
+    double? uMaxRadius,
+    Vector2? uResolutionScale,
+    Vector2? uNoiseScale,
+    double? uNoisePhase,
+    double? uTurbulencePhase,
+    Vector2? uCircle1,
+    Vector2? uCircle2,
+    Vector2? uCircle3,
+    Vector2? uRotation1,
+    Vector2? uRotation2,
+    Vector2? uRotation3,
+  }) {
+    return _program.shader(
+      floatUniforms: Float32List.fromList([
+        ...uColor != null ? uColor.storage : <double>[0, 0, 0, 0],
+        ...uAlpha != null ? <double>[uAlpha] : <double>[0],
+        ...uSparkleColor != null ? uSparkleColor.storage : <double>[0, 0, 0, 0],
+        ...uSparkleAlpha != null ? <double>[uSparkleAlpha] : <double>[0],
+        ...uBlur != null ? <double>[uBlur] : <double>[0],
+        ...uCenter != null ? uCenter.storage : <double>[0, 0],
+        ...uRadiusScale != null ? [uRadiusScale] : <double>[0],
+        ...uMaxRadius != null ? <double>[uMaxRadius] : <double>[0],
+        ...uResolutionScale != null ? uResolutionScale.storage : <double>[0, 0],
+        ...uNoiseScale != null ? uNoiseScale.storage : <double>[0, 0],
+        ...uNoisePhase != null ? <double>[uNoisePhase] : <double>[0],
+        ...uTurbulencePhase != null ? <double>[uTurbulencePhase] : <double>[0],
+        ...uCircle1 != null ? uCircle1.storage : <double>[0, 0],
+        ...uCircle2 != null ? uCircle2.storage : <double>[0, 0],
+        ...uCircle3 != null ? uCircle3.storage : <double>[0, 0],
+        ...uRotation1 != null ? uRotation1.storage : <double>[0, 0],
+        ...uRotation2 != null ? uRotation2.storage : <double>[0, 0],
+        ...uRotation3 != null ? uRotation3.storage : <double>[0, 0],
+      ]),
+    );
+  }
+
+  late ui.FragmentProgram _program;
+
+  /// Direct access to the [ui.FragmentProgram] that this class manages.
+  ///
+  /// In general, this is not needed, but may be useful for debugging or edge cases.
+  ui.FragmentProgram get program => _program;
+
+  /// Direct access to the the SPIR-V bytecode that was used to generate this class.
+  ///
+  /// In general, this is not needed, but may be useful for debugging or edge cases.
+  ///git 
+  /// Words in SPIR-V are 32 bits. Every 4 elements in this list represents 1
+  /// SPIR-V word. See https://www.khronos.org/registry/SPIR-V/.
+  ByteBuffer get spirvByteBuffer => Uint8List.fromList(_spirvByteList).buffer;
+}
+
+/// 11900 bytes
+const List<int> _spirvByteList = <int>[
```

Comment:
The only concern is waiting on async disk load. Happy to put there wherever it is the most painless. The compilation is already async, but I was concerned about having to load this as a web asset.

There is also a ticket to optimize the shader further to make it even less than its current ~11kb

Reply:
The current architecture is to transpile SPIR-V to SkSL, which is then compiled by Skia at runtime. I'm suggesting to adopt the pattern of putting the data in assets to make it easier in the future (e.g. w/ Impeller which won't include a shader compiler) to shift all of the compilation work to AOT when an app is built. So loading an asset now might be a little slower, but it'll make it easier later to shift to a faster approach later.

Reply:
Got it, since this code is currently generated, is it OK to make an issue/TODO, and follow up moving it to an asset in order to get ready for Impeller?

Reply:
https://github.com/flutter/flutter/issues/99783

Reply:
I think it's fine as a follow-up PR, though I'd also like to hear thoughts from others on the reviewers list on compiling in this data. Maybe also @goderbauer

Reply:
It is a bit strange to compile this data in like that. An asset would probably be preferred, e.g. just like we deal with the Material Design Icon font. It should probably not be included if you set "uses-material-design" to false in your pub-spec.

Follow-up PR would be fine, can you reference the issue (https://github.com/flutter/flutter/issues/99783) in a TODO somewhere here?

Reply:
Done